doc/changelog/03-notations/11859-warn-inexact-float.rst

```diff
@@ -0,0 +1,5 @@
+- **Added:**
+  Print a warning when parsing non floating-point values.
```

Comment:
Out of context, this summary might seem a bit too vague. What about:
```suggestion
  In primitive floats, print a warning when parsing a decimal value that is not exactly a floating point number.
```

Reply:
Indeed, thanks!